src/types.ts

```diff
@@ -110,259 +135,134 @@ type IntersectProperties<T> = T extends object
     : never
   : T;
 
```

Comment:
since IntersectProperties is super obscure and black-magickey, please bring back the comment below that demonstrates how it works.
- [ ] bring comment back.

Reply:
Also... It looks like I see a problem - and it might have something to do with some of the slowness issues we saw before - or it might not...
We're assuming that IntersectProperties<{}> will be permissive, but I just tested it in tsplayground and it comes back as "unknown".

We should probably improve IntersectProperties like this...

```
type IntersectProperties<T> = [T] extends [object]
    ? [{}] extends [T] ? {} :
    { [K in keyof T]: (arg: T[K]) => void } extends Record<
        keyof T,
        (arg: infer A) => void
    >
    ? A
    : never
    : {};
```
The differences are the first line (hard-coding what empty-object should return), and the last (simplifying it since we KNOW we're always giving it an object...
Actually, here's an even better option...
```
type IntersectProperties<T extends object> =
    [{}] extends [T] ? {} :
    { [K in keyof T]: (arg: T[K]) => void } extends Record<
        keyof T,
        (arg: infer A) => void
    >
    ? A
    : never;
```
- [ ] little corner case correction and improvement

Reply:
it's done
but one thing
i think this line [{}] extends T can make you think that it's wrong
but i checked it and it works good
when we passing empty object type is {}
if object is not empty it will return what we need
i know that it looks wrong but that's what need's to be done to save the visual studio speed
without that it start to work slower(5 sec to see error when you remove a required param from function call)
[{}] extends [T] or {} extends T will slow it down
[{}] extends T works good and fast